apis/python/src/tiledbsoma/tiledb_group.py

```diff
@@ -53,8 +53,15 @@ def create_unless_exists(self) -> None:
         """
         if not self.exists():
             logger.debug(f"{self._indent}Creating TileDB group {self.nested_name}")
-            tiledb.group_create(uri=self.uri, ctx=self._ctx)
-            self._set_object_type_metadata()
+            try:
+                tiledb.group_create(uri=self.uri, ctx=self._ctx)
+                self._set_object_type_metadata()
+            except tiledb.cc.TileDBError as e:
+                stre = str(e)
+                if (
+                    "already exists" not in stre
+                ):  # This is fine in case of parallel creates
```

Comment:
Formatting and comment here is kind of weird to me. Is this intended to mean something like:

```python
if "already exists" not in str(e):
    # Parallel creation is OK; other cases are genuine errors.
    raise
```